virtManager/details/details.py

```diff
@@ -1110,7 +1111,10 @@ def add_hardware(self, src_ignore):
 
     def remove_xml_dev(self, src_ignore):
         devobj = self.get_hw_row()[HW_LIST_COL_DEVICE]
-        self.remove_device(devobj)
+        from ..delete import vmmDeleteDialog
+        vmmDeleteDialog.set_disk(devobj)
+        vmmDeleteDialog.show_instance(src_ignore, self.vm)
+
```

Comment:
The reason you had to set `topwin` elsewhere is because you are passing `src_ignore` here. You should be passing in `self`

This is going to attempt to launch vmmDeleteDialog for every device removal, not just disk removal. I think a good first patch would be to change this remove_xml_dev function. Have it check `devobj.DEVICE_TYPE == "disk"` and if try, calls a function self.remove_disk. For a first start it should just call self.remove_device(), so nothing changes. All other devices will continue to call self.remove_device(). That could be committed first.

remove_disk() function can then handle initializing vmmDeleteDialog in the right way, and launching it. I don't think it should use show_instance, that's complicating things too much. Just have it do:

```python
dialog = vmmDeleteDialog()
dialog.set_disk(disk)
dialog.show(self.topwin, self.vm)
```